CODING_GUIDELINES.md

```diff
@@ -139,6 +139,28 @@ Temporarily it may be allowed to split unit tests for a module into multiple `te
 
 Temporarily, tests for testing utilities can be placed next to the module containing them, with the name `test_util_<module>.py`. This should be taken as a hint that the tested utils should be moved into the library.
 
+#### Integration Tests Utils
+
+Integrations tests comprise their own utility functions for features' testing, found in `cases.py` for better test automation and simplification:
+
+    - Fields definitions, e.g. IJKField = Field[[IDim, JDim, KDim], np.int64]:
+    - Cases fixtures: cartesian_case for structured, e.g. IDim, JDim; unstructured_case, e.g. EdgeDim.
+    - Parameters allocations to objects, derived directly from the decorator's function definition:
+        ```
+           input_param = cases.allocate(case_fixture, decorator_name, "input_label")()
+        ```
+    - Data initializations/modifications classes, such as ConstInitializer(), to be used as `allocate(...).strategy()` parameters
+    - Decorators verification functions:
+        - `cases.verify()`: used when input and output fields generation cannot be automated. This is for example the case when the `extend()` method needs to be used for an offset or sparse fields are part of the input dataset:
+        - `cases.verify_with_default_data()`: used when input and output fields generation can be automated.
+    - Backends are set automatically with default switched off. However, if explication is needed, they can be explicated as cases attributes.
+
+In case new test utils need to be written, it is important to first check whether any utilities can be use or modified.
+If this is not the case but the new utility supports similar functionalities as others, it should be placed in the same file. Naming convention should mirror other cases and reflect its purpose.
+In case that the new utility is a completely stand-alone component, a new file shall be created in the mutual directory of all usages.
```

Comment:
```suggestion
In case that the new utility is a completely stand-alone component, a new file should be created in the bottom-most directory enclosing all usages of the utility.
```
I'm not sure if this was the intended meaning.